### PR TITLE
front: prevent user from saving multiple times

### LINKED ISSUE #5557: Front: Infra-Editor: Multiple saves allowed.
### What happened?

When we save modifications on a object, the button to save our modifications is still reachable, is not locked, and allow us to save multiple times an object.

### What did you expect to happen?

When an object is being saved, the button shouldn't be reachable, it should be lock (and grey).

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to the Editor
2. Try to edit an object
3. Save it during its save

### What operating system, browser and environment are you using?

- **Browser**: Firefox v119
- **OS**: Nixos v22.11
- **Env**: Dgex Dev

### OSRD version (top right corner `Account` button > `Informations`)

b6936d6f

### PR DESCRIPTION
**Pull request by clarani**
close #5557

### CODE REVIEW

**Review by Wadjetz — APPROVED** [review]

File: front/src/applications/editor/tools/pointEdition/tools.ts [hunk 1]
```diff
@@ -2,10 +2,17 @@ import { BsSkipEnd } from 'react-icons/bs';
...
+  BasePointEditionLayers,
+  SignalEditionLayers,
+} from 'applications/editor/tools/pointEdition/components';
+import getPointEditionTool from 'applications/editor/tools/pointEdition/tool-factory';
+import {
+  getNewBufferStop,
+  getNewDetector,
+  getNewSignal,
+} from 'applications/editor/tools/pointEdition/utils';
+import { BufferStopEntity, DetectorEntity, SignalEntity } from 'types';
```
Comment (anisometropie) [inline front/src/applications/editor/tools/pointEdition/tools.ts]:
```suggestion
import type { BufferStopEntity, DetectorEntity, SignalEntity } from 'types';
```

File: front/src/applications/editor/tools/pointEdition/tools.ts [hunk 2]
```diff
@@ -2,10 +2,17 @@ import { BsSkipEnd } from 'react-icons/bs';
...
 import { FaMapSigns } from 'react-icons/fa';
 
-import getPointEditionTool from './tool-factory';
-import { getNewBufferStop, getNewDetector, getNewSignal } from './utils';
-import { BufferStopEntity, DetectorEntity, SignalEntity } from '../../../../types';
-import { BasePointEditionLayers, SignalEditionLayers } from './components';
+import {
+  BasePointEditionLayers,
+  SignalEditionLayers,
+} from 'applications/editor/tools/pointEdition/components';
```
Comment (anisometropie) [inline front/src/applications/editor/tools/pointEdition/tools.ts]: I don’t see the point to turn these into absolutely path. 

with relative path they make sense together, they are part of the same compoment pointEdition.

if this componet gets moved in the future, each absolute path will have to be change one by one.

imagine doing that for an big component with 20 files, each import part from other files.

File: front/src/applications/editor/tools/pointEdition/tools.ts (hunk 1 above)
Comment (anisometropie) [reply front/src/applications/editor/tools/pointEdition/tools.ts]: this one makes sense in absolute path

File: front/src/applications/editor/tools/rangeEdition/components.tsx [hunk 3]
```diff
@@ -6,29 +6,37 @@ import { BsArrowBarRight } from 'react-icons/bs';
...
 import { FaFlagCheckered, FaTimes } from 'react-icons/fa';
-import CheckboxRadioSNCF from 'common/BootstrapSNCF/CheckboxRadioSNCF';
+
+import EditorContext from 'applications/editor/context';
+import EntityError from 'applications/editor/components/EntityError';
+import EntitySumUp from 'applications/editor/components/EntitySumUp';
+import { NEW_ENTITY_ID } from 'applications/editor/data/utils';
+import { RangeEditionState } from 'applications/editor/tools/rangeEdition/types';
+import EditPSLSection from 'applications/editor/tools/rangeEdition/speedSection/EditPSLSection';
+import { getPointAt, speedSectionIsPsl } from 'applications/editor/tools/rangeEdition/utils';
```
Comment (anisometropie) [inline front/src/applications/editor/tools/rangeEdition/components.tsx]: same remark, every file in `rangeEdition` can import its siblings in relative path

File: front/src/reducers/main/mainSelector.ts [hunk 4]
```diff
@@ -5,6 +5,6 @@ import { MainState } from 'reducers/main/index';
 export const getMain = (state: RootState) => state.main;
 const makeMainSelector = makeSubSelector<MainState>(getMain);
 
-export const getLoading = makeMainSelector('loading');
+export const getIsLoading = (state: RootState) => state.main.loading > 0;
```
Comment (anisometropie) [inline front/src/reducers/main/mainSelector.ts]:
```suggestion
export const getLoading = makeMainSelector('loading');
export const getIsLoading = (state: RootState) => getLoading(state) > 0;
```

**Review by anisometropie — CHANGES_REQUESTED** [review]

File: front/src/applications/editor/tools/pointEdition/tools.ts (hunk 2 above)
Comment (clarani) [reply front/src/applications/editor/tools/pointEdition/tools.ts]: Ok, I got your point and I can change that :)

File: front/src/applications/editor/tools/pointEdition/tools.ts (hunk 1 above)
Comment (clarani) [reply front/src/applications/editor/tools/pointEdition/tools.ts]: Done

File: front/src/applications/editor/tools/rangeEdition/components.tsx (hunk 3 above)
Comment (clarani) [reply front/src/applications/editor/tools/rangeEdition/components.tsx]: Done

File: front/src/reducers/main/mainSelector.ts (hunk 4 above)
Comment (clarani) [reply front/src/reducers/main/mainSelector.ts]: Done :+1:

File: front/src/applications/editor/tools/switchEdition/components.tsx [hunk 5]
```diff
@@ -8,32 +8,36 @@ import { featureCollection, point } from '@turf/helpers';
...
-import { getMap } from '../../../../reducers/map/selectors';
-import { ExtendedEditorContextType } from '../editorContextTypes';
-import { CustomSchemaField } from './components/CustomSchemaField';
-import useSwitch from './useSwitch';
+  FlatSwitchEntity,
+  flatSwitchToSwitch,
+  getNewSwitch,
+  isSwitchValid,
+} from 'applications/editor/tools/switchEdition/utils';
+import { SwitchEditionState } from 'applications/editor/tools/switchEdition/types';
```
Comment (anisometropie) [inline front/src/applications/editor/tools/switchEdition/components.tsx]: This one too can stay relative, it’s the same component in the same folder

File: front/src/applications/editor/tools/switchEdition/components.tsx [hunk 6]
```diff
@@ -8,32 +8,36 @@ import { featureCollection, point } from '@turf/helpers';
...
-import { getInfraID } from '../../../../reducers/osrdconf/selectors';
-import { getMap } from '../../../../reducers/map/selectors';
-import { ExtendedEditorContextType } from '../editorContextTypes';
-import { CustomSchemaField } from './components/CustomSchemaField';
-import useSwitch from './useSwitch';
+  FlatSwitchEntity,
+  flatSwitchToSwitch,
+  getNewSwitch,
+  isSwitchValid,
+} from 'applications/editor/tools/switchEdition/utils';
```
Comment (anisometropie) [inline front/src/applications/editor/tools/switchEdition/components.tsx]: This one too can stay relative

File: front/src/applications/editor/tools/trackEdition/components.tsx [hunk 7]
```diff
@@ -6,30 +6,36 @@ import { last } from 'lodash';
...
 import { BsBoxArrowInRight } from 'react-icons/bs';
 
+import EditorForm from 'applications/editor/components/EditorForm';
+import EntitySumUp from 'applications/editor/components/EntitySumUp';
+import EntityError from 'applications/editor/components/EntityError';
+import { getEntities } from 'applications/editor/data/api';
+import { NEW_ENTITY_ID } from 'applications/editor/data/utils';
+import EditorContext from 'applications/editor/context';
+import { TrackEditionState } from 'applications/editor/tools/trackEdition/types';
+import { injectGeometry } from 'applications/editor/tools/trackEdition/utils';
```
Comment (anisometropie) [inline front/src/applications/editor/tools/trackEdition/components.tsx]: here too

File: front/src/applications/editor/tools/switchEdition/components.tsx (hunk 6 above)
Comment (clarani) [reply front/src/applications/editor/tools/switchEdition/components.tsx]: Done

File: front/src/applications/editor/tools/trackEdition/components.tsx (hunk 7 above)
Comment (clarani) [reply front/src/applications/editor/tools/trackEdition/components.tsx]: Done

File: front/src/applications/editor/tools/switchEdition/components.tsx (hunk 5 above)
Comment (clarani) [reply front/src/applications/editor/tools/switchEdition/components.tsx]: Done

**Review by SharglutDev — APPROVED** [review]
Lgtm !

**Review by Math-R — APPROVED** [review]

**Review by anisometropie — APPROVED** [review]

**Pull request merged**